evalml/automl/automl_search.py

```diff
@@ -645,7 +641,13 @@ def _compute_cv_scores(self, pipeline, X, y):
         start = time.time()
         cv_data = []
         logger.info("\tStarting cross validation")
-        for i, (train, test) in enumerate(self.data_split.split(X, y)):
+        X = _convert_to_woodwork_structure(X)
+        y = _convert_to_woodwork_structure(y)
+
+        X_pd = _convert_woodwork_types_wrapper(X.to_dataframe())
+        y_pd = _convert_woodwork_types_wrapper(y.to_series())
+        for i, (train, test) in enumerate(self.data_split.split(X_pd, y_pd)):
```

Comment:
Do we need to convert to pandas? I think data tables are splitable:

![image](https://user-images.githubusercontent.com/41651716/99853983-f7858480-2b51-11eb-8a67-b12ab5105e30.png)

Reply:
@freddyaboulton KFolds works, but I don't think the other ones do (Stratified):

I think because of the attributes on/missing from DataTables, scikit-learn's `check_array` fails:

```
evalml/automl/automl_search.py:649: in _compute_cv_scores
    for i, (train, test) in enumerate(self.data_split.split(X, y)):
../venv/lib/python3.7/site-packages/sklearn/model_selection/_split.py:735: in split
    y = check_array(y, ensure_2d=False, dtype=None)
../venv/lib/python3.7/site-packages/sklearn/utils/validation.py:72: in inner_f
    return f(**kwargs)
../venv/lib/python3.7/site-packages/sklearn/utils/validation.py:648: in check_array
    n_samples = _num_samples(array)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

x = array(<DataColumn: None (Physical Type = Int64) (Logical Type = WholeNumber) (Semantic Tags = {'numeric'})>,
      dtype=object)

    def _num_samples(x):
        """Return number of samples in array-like x."""
        message = 'Expected sequence or array-like, got %s' % type(x)
        if hasattr(x, 'fit') and callable(x.fit):
            # Don't get num_samples from an ensembles length!
            raise TypeError(message)

        if not hasattr(x, '__len__') and not hasattr(x, 'shape'):
            if hasattr(x, '__array__'):
                x = np.asarray(x)
            else:
                raise TypeError(message)

        if hasattr(x, 'shape') and x.shape is not None:
            if len(x.shape) == 0:
                raise TypeError("Singleton array %r cannot be considered"
>                               " a valid collection." % x)
E               TypeError: Singleton array array(<DataColumn: None (Physical Type = Int64) (Logical Type = WholeNumber) (Semantic Tags = {'numeric'})>,
E                     dtype=object) cannot be considered a valid collection.

../venv/lib/python3.7/site-packages/sklearn/utils/validation.py:196: TypeError
```

Reply:
Ah thanks for that @angela97lin ! I forgot about the stratified split 😂 

The issue is that [deep-inside sklearn](https://github.com/scikit-learn/scikit-learn/blob/0fb307bf39bbdacd6ed713c00724f8f871d60370/sklearn/utils/validation.py#L598), it calls `np.asarray` on the data column. This produces a weird result that doesn't have a meaningful shape. Totally cool to keep this as-is! Just wanted to make sure we weren't calling `to_series` if we didn't need to!